packages/core/src/Renderer.ts

```diff
@@ -329,6 +339,51 @@ export class Renderer extends AbstractRenderer
         this.resize(this.options.width, this.options.height);
     }
 
+    protected contextChange(): void
+    {
+        const gl = this.gl;
+
+        let samples;
+
+        if (this.context.webGLVersion === 1)
+        {
+            const framebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);
+
+            gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+
+            samples = gl.getParameter(gl.SAMPLES);
+
+            gl.bindFramebuffer(gl.FRAMEBUFFER, framebuffer);
+        }
+        else
+        {
+            const framebuffer = gl.getParameter(gl.DRAW_FRAMEBUFFER_BINDING);
+
+            gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, null);
```

Comment:
Bit of duplicated code here! should we wrap the if WebGLVersion around this line only?

Reply:
`DRAW_FRAMEBUFFER` and `DRAW_FRAMEBUFFER_BINDING` are webGL2 only, and we don't want to touch the read buffer with webGL2. We would have to wrap around the second bind as well, but that would be more lines of code:

```js
let framebuffer;

if (this.context.webGLVersion === 1)
{
    framebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);

    gl.bindFramebuffer(gl.FRAMEBUFFER, null);
}
else
{
    framebuffer = gl.getParameter(gl.DRAW_FRAMEBUFFER_BINDING);

    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, null);
}

samples = gl.getParameter(gl.SAMPLES);

if (this.context.webGLVersion === 1)
{
    gl.bindFramebuffer(gl.FRAMEBUFFER, framebuffer);
}
else
{
    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, framebuffer);
}
```

Maybe like this assuming the constants would be undefined in webGL1:

```js
const framebuffer = gl.getParameter(gl.DRAW_FRAMEBUFFER_BINDING ?? gl.FRAMEBUFFER_BINDING);

gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER ?? gl.FRAMEBUFFER, null);

samples = gl.getParameter(gl.SAMPLES);

gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER ?? gl.FRAMEBUFFER, framebuffer);
```